src/wasm/wasm.cpp

```diff
@@ -1590,6 +1608,13 @@ void Module::removeTags(std::function<bool(Tag*)> pred) {
   removeModuleElements(tags, tagsMap, pred);
 }
 
+void Module::updateDataSegmentsMap() {
+  dataSegmentsMap.clear();
+  for (auto& curr : dataSegments) {
+    dataSegmentsMap[curr->name] = curr.get();
+  }
+}
```

Comment:
We don't seem to have this for ElementSegments? Why is this not symmetric to that?

Reply:
It's a bug that I never updated the maps when writing the PR for First Class Data Segments. In the future, I'd make this a PR that comes before multi-memories, and it still can!

This is not symmetric to ElementSegments, because we don't do vector swaps for ElementSegments like we do for DataSegments: [Ex1](https://sourcegraph.com/github.com/WebAssembly/binaryen@multi-memories/-/blob/src/ir/memory-utils.h?L165:15&subtree=true) [Ex2](https://sourcegraph.com/github.com/WebAssembly/binaryen@multi-memories/-/blob/src/passes/MemoryPacking.cpp?L516:11&subtree=true)

I thought it might be wasteful to call updateMaps() just for DataSegments, but there is precedent for doing so with [Functions](https://sourcegraph.com/github.com/WebAssembly/binaryen@multi-memories/-/blob/src/passes/opt-utils.h?L39:6&subtree=true). Happy to remove this function and just call updateMaps(), wdyt?

Reply:
Hmm, I get 404s on both the first 2 links?

Reply:
Makes sense, the multi-memories branch is deleted now. Fixed the links to point to main now. [Ex1](https://sourcegraph.com/github.com/WebAssembly/binaryen@main/-/blob/src/ir/memory-utils.h?L165&subtree=true) [Ex2](https://sourcegraph.com/github.com/WebAssembly/binaryen@main/-/blob/src/passes/MemoryPacking.cpp?L176&subtree=true)

Reply:
I see, thanks! Yeah, maybe it's worth having a function for it, if we have multiple places that don't need to update all the maps. lgtm as it is then.